### PR TITLE
Alter cruise speed at any time during a mission.

### PR DESCRIPTION
**Pull request by SimonWilks**
Sending the MAV_CMD_DO_CHANGE_SPEED at anytime during a mission will result in the vehicle's current cruise speed being changed to the newly set value immediately. This provides smooth speed control throughout the allowed range for the vehicle type.

@AndreasAntener want to take a look? I can flight test this on multirotor next week. Works nicely on SITL.

### CODE REVIEW

**Comment by AndreasAntener** [comment]
I just realized that this is a FW PR ;) will have a look tomorrow

**Comment by SimonWilks** [comment]
@AndreasAntener are we able to take this across to our private firmware or should we try to get it into PX4?

File: src/modules/navigator/navigator_main.cpp [hunk 1]
```diff
@@ -684,17 +684,16 @@ Navigator::get_cruising_speed()
 {
 	/* there are three options: The mission-requested cruise speed, or the current hover / plane speed */
 	if (_vstatus.is_rotary_wing) {
-		if (_mission_cruising_speed_mc  > FLT_EPSILON
+		if (_mission_cruising_speed_mc  > 0.0f
```
Comment (AndreasAntener) [inline src/modules/navigator/navigator_main.cpp:687]: Sorry to be a bit late here, but why was this needed?

**Comment by AndreasAntener** [comment]
I'm taking it over to ours and push the commit to my mission sync branch for PX4 (because all the dependencies for this are already there)

File: src/modules/navigator/navigator_main.cpp (hunk 1 above)
Comment (SimonWilks) [reply src/modules/navigator/navigator_main.cpp:687]: If I recall correctly there aren't floating calculations done on this variable, it's just the compiler checks forcing the warnings for equality tests. For < or > tests we aren't ever floating around near zero values so just doing a normal check is fine. But other than that, no need for the change.

**Comment by AndreasAntener** [comment]
Thanks, cherry-picked
